core/src/main/java/org/elasticsearch/index/get/GetField.java

```diff
@@ -90,4 +95,58 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeGenericValue(obj);
         }
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startArray(name);
+        for (Object value : values) {
+            builder.value(value);
+        }
+        builder.endArray();
+        return builder;
+    }
+
+    public static GetField fromXContent(XContentParser parser) throws IOException {
+        if (parser.currentToken() != XContentParser.Token.FIELD_NAME) {
+            throw new ParsingException(parser.getTokenLocation(),
+                    "expected " + XContentParser.Token.FIELD_NAME + " - found " + parser.currentToken());
+        }
+        String fieldName = parser.currentName();
+        List<Object> values = new ArrayList<>();
+        XContentParser.Token token = parser.nextToken();
+        if (token != XContentParser.Token.START_ARRAY) {
+            throw new ParsingException(parser.getTokenLocation(),
+                    "expected " + XContentParser.Token.START_ARRAY + " - found " + parser.currentToken());
+        }
+        while(parser.nextToken() != XContentParser.Token.END_ARRAY) {
+            values.add(parser.objectText());
```

Comment:
What' really strange is that stored `geo_point` fields get printed as a flat array, so if you store two point like:
```
PUT twitter/tweet/1
{
  "counter": 1,
  "location": [
    {
      "lat": 0.45,
      "lon": 0.33
    },
    {
      "lon": 1,
      "lat": 1.5
    }
  ]
}
```
I get back:
```
GET twitter/tweet/1?stored_fields=location

{
  "_index": "twitter",
  "_type": "tweet",
  "_id": "1",
  "_version": 4,
  "found": true,
  "fields": {
    "location": [
      "0.45, 0.33",
      "1.5, 1.0"
    ]
  }
}
```

So it looks like its printed as array of plain values. From reading  `ContentBuilder#Writers -> #value(GeoPoint p) -> #latlon()`  would have expected an array of objects, each containing a lat/lon parameters. It seems tricky to see how these values actually get stored in the index and then written to the response, so its also hard to parse them? I'd at least add the option of parsing maps here, then I think we are on the safe side.